File: eAnimalShelter/UI/eanimalshelter_mobile/lib/models/favorite.dart
import 'animal.dart';

class Favorite {
  final int favoriteId;
  final Animal animal;
  final DateTime dateAdded;

  Favorite({
    required this.favoriteId,
    required this.animal,
    required this.dateAdded,
  });

  factory Favorite.fromJson(
      Map<String, dynamic> json) {
    return Favorite(
      favoriteId: json["favoriteId"],
      animal: Animal.fromJson(json["animal"]),
      dateAdded:
          DateTime.parse(json["dateAdded"]),
    );
  }
}

File: eAnimalShelter/eAnimalShelter.Services/Database/Favorite.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eAnimalShelter.Services.Database
{
    public class Favorite
    {
        [Key]
        public int FavoriteId { get; set; }
        public int AnimalId { get; set; }
        [ForeignKey("AnimalId")]
        public Animal Animal { get; set; } = null!;
        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; } = null!;
        public DateTime DateAdded { get; set; }
    }
}